### ISSUE TITLE
Add Galaxy API

### ISSUE THREAD

**Issue by Odelpasso** [sharkwouter/minigalaxy#175]
Hello,

I do not participate since a long time to MiniGalaxy and i would like to know if you (@sharkwouter) can implement the possibility to download the game directly instead of the installer (.exe or .sh)

Currently, lgogdownloader do that and i can download without problem game like on Galaxy 2.0.

You can find Galaxy integration python here : 
https://github.com/gogcom/galaxy-integrations-python-api

With this doc, you can even add the achievements ... 

Thanks,
Odel

**Comment by sharkwouter** [comment]
Thanks for the contribution, sorry for replying so late. This sounds good. Could you explain it a bit more? What does the downloaded content look like?

**Comment by Odelpasso** [comment]
Hello,

The main difference is that when you use GoG API, it download the setup (.sh for Linux or .exe/.bin for Windows).
But when you download with Galaxy API like lgogdownloader, you download directly the game like when you download a game on Steam/UPlay/Origin etc...

**Comment by arrowgent** [comment]
ah hmm.  i think if youre going to allow this then it should be an option

i may prefer to download the setup file .sh or .exe for storage or transferring to a different computer

but i understand that instead of downloading a package (50mb up to 50GB) Then unpacking/extracting is excessive
a compressed package then extracting is both a size and time waste
double the size, and a small % of cpu time used to decompress the file

so youre saying the GoG API has a method to download exactly the extracted packages for games? just like steam?
are you sure steam are not packages which are extracted/decompressed on the fly?

**Comment by Odelpasso** [comment]
For Steam, i'm not sure how it works but it's not packages like GoG.
You can check/find galaxy used by lgogdownloader on their github https://github.com/Sude-/lgogdownloader

**Comment by Archehandoro** [comment]
Achievements and time tracked is nice feature, but much like @arrowgent I don't see the benefit for downloading he game files directly as opposed to .sh/.exe binaries. It might even conflict with how Wine interacts with the games too.

**Comment by spannerman79** [comment]
No this wouldn't work for Linux native games.

For example - using lgogdownloader;

```
lgogdownloader --list --game neverwinter
Getting game names (4/4) 14 / 14
neverwinter_nights_2_complete
neverwinter_nights_diamond_edition
neverwinter_nights_enhanced_edition
```

Now thats what is on my account.  And I know for a **fact** that NWN E **is** on Linux.  So lets see if your suggestion @Odelpasso will work using the Galaxy API.

```
lgogdownloader --galaxy-platform linux --galaxy-show-builds neverwinter_nights_enhanced_edition
Getting game names (4/4) 14 / 14
Galaxy API doesn't have Linux support
Checking for installers that can be used as repository
Using these installers
        neverwinter_nights_enhanced_edition/en3installer0
        neverwinter_nights_wyvern_crown_of_cormyr/en3installer0
        neverwinter_nights_infinite_dungeons/en3installer0
        neverwinter_nights_pirates_of_the_sword_coast/en3installer0
        neverwinter_nights_heroes_of_neverwinter/en3installer0
        neverwinter_nights_darkness_over_daggerford/en3installer0
        neverwinter_nights_tyrants_of_the_moonsea/en3installer0
        neverwinter_nights_dark_dreams_of_furiae/en3installer0
```

Nope - Galaxy API is **useless** for Linux support.

**Comment by Odelpasso** [comment]
Before to complain it does not work, try it please :
![Capture d’écran de 2021-03-12 21-08-01](https://user-images.githubusercontent.com/3606036/110993232-50825e00-8377-11eb-9047-4bc811a8d9d9.png)

I download torchlight 2 linux with galaxy API

**Comment by spannerman79** [comment]
> Before to complain it does not work, try it please :

My **actual** comment proved that I just did & showed. _Ugh_.

Secondly - are you going to pay those that have limited conenctiones a higher download allotment to download invidual files vs packed compressed `.sh` archives?  I seriously doubt it.

Think **and read** before making innate comments.

**Comment by Odelpasso** [comment]
> My **actual** comment proved that I just did & showed. _Ugh_.
Your previous command does not prove anything, you didn't try to install the game ...

> Secondly - are you going to pay those that have limited conenctiones a higher download allotment to download invidual files vs packed compressed `.sh` archives? I seriously doubt it.
Do you know that it's possible to add an option for the user like in Galaxy, download the game or the offline installer ?

> Think and read before making innate comments.
And you, try to understand how works lgogdownloader... 
**"No this wouldn't work for Linux native games."**  --> It's the case and i proved that.

**Comment by spannerman79** [comment]
> Your previous command does not prove anything, you didn't try to install the game ...

Yes I did, not my fault that you couldn't guess that EOL was actual EOL in console.



> Do you know that it's possible to add an option for the user like in Galaxy, download the game or the offline installer ?

Now you are changing tact - common thing that people that do strawmanning.

I'm done replying to you.  Blocked.

**Comment by Bitwolfies** [comment]
Achievements would be killer, the only way to get them currently on linux is to run the actual gog clients through wine. The options to download the files directly would be a preferred option for me, dont need twice the storage of what you're downloading and will work just fine with wine. (As this is what steam proton does)

**Comment by imLinguin** [comment]
Galaxy API is only useful for Windows Games, I implemented that from scratch in [my project](https://github.com/imLinguin/dvd-projekt). I would like to contribute to mingalaxy, after testing this solution for more games.

**Comment by sharkwouter** [comment]
Contributions to make this work would be more than welcome :)
